File: Korero/Models/Thread.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Korero.Models
{
    public class Thread
    {
        public int ID { get; set; }

        [Required]
        [StringLength(64)]
        public string Title { get; set; }

        public DateTime DateCreated { get; set; }
        public int Views { get; set; }

        public ApplicationUser Author { get; set; }
        public List<Reply> Replies { get; set; }
        public Tag Tag { get; set; }
    }
}

File: Korero/ClientApp/app/models/thread.ts
import { Reply } from './reply'
import { Tag } from './tag'
import { User } from './user'

export class Thread {
    public id: number;
    public title: string;
    public dateCreated: Date;
    public views: number;
    public tag: Tag;
    public replies: Reply[];
    public author: User;

    public show: boolean = true; // Filtering purposes ;)
}

export class ThreadData {
    public total: number;
    public data: Thread[];
}